js/components/main.js

```diff
@@ -620,6 +623,16 @@ class Main extends ImmutableComponent {
     windowActions.setUrlBarActive(false)
   }
 
+  onFindHide () {
+    const activeFrame = FrameStateUtil.getActiveFrame(this.props.windowState)
+    windowActions.setFindbarShown(activeFrame, false)
+    webviewActions.stopFindInPage()
```

Comment:
@bridiver this had to be called here to retain the existing behaviour. 
`stopFindInPage` needs to be called to clear the highlight but to keep the current match in case you `command+g`.